rfcs/proposed/core_types/README.md

```diff
@@ -318,34 +300,371 @@ is added, a corresponding `set_numa_ids` function would likely follow. The choic
+  as an opaque type. That proposal therefore relies on (and, again, exposes) implementation details. It might be
+  acceptable, as changing these details would likely break backward compatibility anyway. But perhaps we can
+  do better and avoid such exposure at all.
+- It mostly sticks to the current usage model of `constraints`, possibly missing an opportunity to simplify
+  the API usage for both basic and advanced scenarios.
+
+### New API (sketched)
+
+#### Header
 
 ```cpp
-bool has_core_type(core_type_id id) const;
-size_t num_core_types() const;
+#include <oneapi/tbb/task_arena.h>
 ```
 
-4. **Builder Pattern Extensions**: Should we support incremental building?
+#### Synopsis
 
 ```cpp
-constraints& add_core_type(core_type_id id);
-constraints& remove_core_type(core_type_id id);
-void clear_core_types();
+namespace oneapi {
+namespace tbb {
+class task_arena {
+  public:
+    // (Possibly needed) New constant to indicate a selectable constraint
+    static constexpr int selectable = /* unspecified */;
+    
+    // New constructor template
+    template <typename Selector>
+    task_arena(constraints a_constraints, Selector a_selector, unsigned reserved_slots = 1,
+               priority a_priority = priority::normal);
+
+    // New template for overloads of initialize
+    template <typename Selector>
+    /*return type to be defined*/
+    initialize(constraints a_constraints, Selector a_selector, unsigned reserved_slots = 1,
+               priority a_priority = priority::normal);
+
+};
+}}
 ```
 
-5. **Core Types Data member**: Should we store core types in a new container data member instead of bit-packing?
-   - Pros: Simpler logic, easier to extend
-   - Cons: Increases struct size, breaks ABI compatibility
+#### API description
+
+The new templates for construction and initialization of a `task_arena` accept, in addition to `constraints`,
+a *selection function*: a user-specified callable object that ranks core types or NUMA nodes available on the platform.
+
+We can consider at least two variations of what the selection function accepts as its arguments:
+1. the whole vector of core types or NUMA nodes, as returned by `tbb::info` API calls, or
+2. a single `core_type_id` or `noma_node_id` value, packed into a tuple/struct with additional useful information,
+   specifically the number of entities to choose from (e.g., `tbb::info::core_types().size()`) and the position
+   (index) of the given ID value.
+
+In principle, it could be possible for the implementation to support both variations, distinguished at compile time.
+For each ID value, the selection function should return a signed integral number as the score for that ID. In case (1),
+the function should therefore return a vector of scores, one for each element of its input vector.
+
+A negative score would indicate that the corresponding core type or NUMA node should be excluded from use
```

Comment:
Perhaps, for better readability the negative score could also have their own name such as `skip_resource`, for example.